### ISSUE TITLE
Amazon Echo Control Binding leaks threads

### ISSUE THREAD

**Issue by boc-tothefuture** [openhab/openhab-addons#7977]
I have had several issues over the past few days where my openhab system stopped processing rules.  

I would receive the following in the log, when attempting to connect via the openhab-cli console and when rules needed to fire:

```
java.lang.OutOfMemoryError: unable to create new native thread
```

I also had this error message multiple times in my logs:
```
2020-06-21 14:50:49.806 [INFO ] [control.internal.WebSocketConnection] - Web Socket error
java.nio.channels.AsynchronousCloseException: null
        at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.close(HttpConnectionOverHTTP.java:181) ~[?:?]
        at java.util.ArrayList.forEach(ArrayList.java:1257) [?:1.8.0_201]
        at org.eclipse.jetty.client.AbstractConnectionPool.close(AbstractConnectionPool.java:208) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.DuplexConnectionPool.close(DuplexConnectionPool.java:237) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpDestination.close(HttpDestination.java:385) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.client.HttpClient.doStop(HttpClient.java:260) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:93) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.component.ContainerLifeCycle.stop(ContainerLifeCycle.java:180) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStop(ContainerLifeCycle.java:201) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.websocket.client.WebSocketClient.doStop(WebSocketClient.java:371) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.component.AbstractLifeCycle.stop(AbstractLifeCycle.java:93) [bundleFile:9.4.20.v20190813]
        at org.openhab.binding.amazonechocontrol.internal.WebSocketConnection.close(WebSocketConnection.java:159) [bundleFile:?]
        at org.openhab.binding.amazonechocontrol.internal.WebSocketConnection$2.run(WebSocketConnection.java:184) [bundleFile:?]
        at java.util.TimerThread.mainLoop(Timer.java:555) [?:1.8.0_201]
        at java.util.TimerThread.run(Timer.java:505) [?:1.8.0_201]
```

Once I restarted, I created a few dev-dumps and took a look at the threads:
With the Amazon Echo Control Binding running:
Right after startup:
```
cat threads.txt | grep "Id=" | sort | grep -i websocket | wc -l                                   
9
```

15 minutes later:
```
cat threads.txt | grep "Id=" | sort | grep -i websocket | wc -l                              
153
```

Another 15 minutes later:

```
cat threads.txt | grep "Id=" | sort | grep -i websocket | wc -l                                 
289
```

I tried disabling a bunch of different bindings.  Once I disabled the Amazon Echo Control binding the thread leaks went away.

**Comment by lolodomo** [comment]
What OH version are you running? 
It should be fixed in 2.5.6 released yesterday.

**Comment by boc-tothefuture** [comment]
@lolodomo 2.5.5.  I didn't realize this was an identified, closed defect.
